nixos/modules/programs/openvpn3.nix

```diff
@@ -1,43 +1,116 @@
+      type = submodule {
+        options = {
+          settings = mkOption {
+            description =
+              "Options stored in {file}`/etc/openvpn3/netcfg.json` configuration file";
+            default = { };
+            type = submodule {
+              freeformType = json.type;
+              options = {
+                systemd_resolved = mkOption {
+                  type = bool;
+                  description = "Whether to use systemd-resolved integration";
+                  default = config.services.resolved.enable;
+                  defaultText =
+                    literalExpression "config.services.resolved.enable";
+                  example = false;
+                };
+              };
+            };
+          };
+        };
+      };
+    };
+    log-service = mkOption {
+      description = "Log service configuration";
+      default = { };
+      type = submodule {
+        options = {
+          settings = mkOption {
+            description =
+              "Options stored in {file}`/etc/openvpn3/log-service.json` configuration file";
+            default = { };
+            type = submodule {
+              freeformType = json.type;
+              options = {
+                journald = mkOption {
+                  description = "Use systemd-journald";
+                  type = bool;
+                  default = true;
+                  example = false;
+                };
+                log_dbus_details = mkOption {
+                  description = "Add D-Bus details in log file/syslog";
+                  type = bool;
+                  default = true;
+                  example = false;
+                };
+                log_level = mkOption {
+                  description = "How verbose should the logging be";
+                  type = (ints.between 0 7) // {
+                    merge = _loc: defs:
+                      lists.foldl max 0 (options.getValues defs);
+                  };
+                  default = 3;
+                  example = 6;
+                };
+                timestamp = mkOption {
+                  description = "Add timestamp log file";
+                  type = bool;
+                  default = false;
+                  example = true;
+                };
+              };
+            };
+          };
+        };
       };
-      defaultText = lib.literalExpression ''pkgs.openvpn3.override {
-        enableSystemdResolved = config.services.resolved.enable;
-      }'';
-      description = ''
-        Which package to use for `openvpn3`.
-      '';
     };
   };
 
   config = lib.mkIf cfg.enable {
-    services.dbus.packages = [
-      cfg.package
-    ];
+    services.dbus.packages = [ cfg.package ];
 
     users.users.openvpn = {
       isSystemUser = true;
       uid = config.ids.uids.openvpn;
       group = "openvpn";
     };
 
-    users.groups.openvpn = {
-      gid = config.ids.gids.openvpn;
+    users.groups.openvpn = { gid = config.ids.gids.openvpn; };
```

Comment:
Do we really need to use static gids here?

Reply:
Don't think that it is a requirement now, but since it was part of the previous module version, I kept it

Reply:
I guess we are breaking backwards compat anyway, since NixOS rebuild will fail due to the old config file (generated by the binary) already existing, preventing NixOS creating the symlink to NixOS-generated file. And since backwards compat is broken already, we might as well stop using static gid (unless dbus needs static).

Speaking of backwards compat, we probably want to add this notice (as well as the info about the module and binary being updated) to the release notes.

Reply:
> since NixOS rebuild will fail due to the old config file (generated by the binary) already existing, preventing NixOS creating the symlink to NixOS-generated file

Actually, it won't since the default path is different from `/etc/openvpn3` (to which I do the explicit override when building the package now) thus the updating clients will simply have the new config path without any effect on the old one.

Reply:
As for now, the ID is kept to maintain (some) backwards compatibility although a future PR may come later with the removal of this dependency on static IDs